workflow/rules/DNA.smk

```diff
@@ -0,0 +1,453 @@
+    Finally, FASTQC is used to generate quality reports of the raw and final read sets in order to assess read quality improvement. 
+    The users have control over which of the above features they wish to use.
+
+    Quality-control step accomplishes three tasks:
+        - Trims adapter sequences from input read pairs (.fastq.gz)
+        - Quality filters from input read pairs
+        - Removes human contamination from input read pairs
+
+    @Container requirements
+        - [MetaWrap](https://github.com/bxlab/metaWRAP) orchestrates execution of 
+            these tools in a psuedo-pipeline:
+            - [FastQC](https://github.com/s-andrews/FastQC)
+            - [bmtagger](ftp://ftp.ncbi.nlm.nih.gov/pub/agarwala/bmtagger/)
+            - [Trim-Galore](https://github.com/FelixKrueger/TrimGalore)
+            - [Cutadapt](https://github.com/marcelm/cutadapt)
+    
+    @Environment specifications
+        - minimum 16gb memory
+        - will load bmtagger index in memory (8gb)
+
+    @Input:
+        - Raw fastq reads (R1 & R2 per-sample) from the instrument
+        
+    @Outputs:
+        - Trimmed, quality filtered reads (R1 & R2)
+        - FastQC html report and zip file on trimmed data
+    """
+    input:
+        R1                  = start_r1,
+        R2                  = start_r2,
+    output:
+        R1_pretrim_report   = join(top_readqc_dir, "{name}", "{name}_R1_pretrim_report.html"),
+        R2_pretrim_report   = join(top_readqc_dir, "{name}", "{name}_R2_pretrim_report.html"),
+        R1_postrim_report   = join(top_readqc_dir, "{name}", "{name}_R1_postrim_report.html"),
+        R2_postrim_report   = join(top_readqc_dir, "{name}", "{name}_R2_postrim_report.html"),
+        R1_trimmed          = join(top_trim_dir, "{name}", "{name}_R1_trimmed.fastq"),
+        R2_trimmed          = join(top_trim_dir, "{name}", "{name}_R2_trimmed.fastq"),
+        R1_trimmed_gz       = join(top_trim_dir, "{name}", "{name}_R1_trimmed.fastq.gz"),
+        R2_trimmed_gz       = join(top_trim_dir, "{name}", "{name}_R2_trimmed.fastq.gz"),
+    params:
+        rname               = "metawrap_read_qc",
+        sid                 = "{name}",
+        this_qc_dir         = join(top_readqc_dir, "{name}"),
+        trim_out            = join(top_trim_dir, "{name}"),
+        tmpr1               = lambda _, output, input: join(config['options']['tmp_dir'], str(basename(str(input.R1))).replace('_R1.', '_1.').replace('.gz', '')),
+        tmpr2               = lambda _, output, input: join(config['options']['tmp_dir'], str(basename(str(input.R2))).replace('_R2.', '_2.').replace('.gz', '')),
+    containerized: metawrap_container,
+    threads: int(cluster["metawrap_genome_assembly"].get('threads', default_threads)),
+    shell: 
+        """
+            # uncompress to lscratch
+            rone="{input.R1}"
+            ext=$(echo "${{rone: -2}}" | tr '[:upper:]' '[:lower:]')
+            if [[ "$ext" == "gz" ]]; then
```

Comment:
both `SLURM_JOBID` and `SLURM_JOB_ID` are valid for slurm. `SLURM_JOBID` only exists for[ backwards compatibility](https://slurm.schedmd.com/sbatch.html#OPT_SLURM_JOBID). I think metamorph uses `SLURM_JOBID` through out.

If this is saved to lscratch does it need a temp dir and/or a trap, it's non persistent disk space that will get blown away once the jobs completes?

Reply:
Yeah, there is a mechanism to override the location. That is what the `--tmp-dir` option of the pipeline.  If some one did use that option then the files would not get cleaned up. With that being said, it is extremely important to always create a temp directory inside of that location, I would just use mktemp command. The idea behind that is to create a sandbox for where the files get generated to avoid collisons due to naming and to prevent anything bad from happening when you do delete the directory.

Here is a basic example of it being used in the wild:
https://github.com/OpenOmics/genome-seek/blob/e03ded4fd0fdb975fed60df28c35d34c8f9c42c5/workflow/rules/qc.smk#L69

`${{tmp}}` will resolve to the absolute path of the temp directory created by mktemp.